core/modules/wpublish/ResourceMonitor.cc

```diff
@@ -78,16 +67,8 @@ unsigned int ResourceMonitor::count(int chunk,
 }
 
 
-ResourceMonitor::ResourceCounter ResourceMonitor::resourceCounter() const {
-
-    // Make a copy of the map while under protection of the lock guard.
-    lock_guard<mutex> lock(_mtx);
-    ResourceCounter result = _resourceCounter;
-    return result;
-}
-
-
 json ResourceMonitor::statusToJson() const {
+    lock_guard<mutex> lock(_mtx);
     json result = json::array();
     for (auto&& entry: _resourceCounter) {
         result.push_back({entry.first, entry.second});
```

Comment:
I'm guessing that the race condition was caused by another thread altering _resourceCounter while in the loop.

Reply:
Exactly!